Sources/protoc-gen-swift/OneofGenerator.swift

```diff
@@ -49,6 +50,53 @@ class OneofGenerator {
         self.swiftRelativeName = sanitizeOneofTypeName(descriptor.swiftRelativeType)
         self.swiftFullName = swiftMessageFullName + "." + swiftRelativeName
         self.comments = file.commentsFor(path: path)
+
+        let first = fieldsSortedByNumber.first!.number
+        let last = fieldsSortedByNumber.last!.number
+        // Easy case, all in order and no gaps:
+        if first + fields.count - 1 == last {
+            oneofIsContinuousInParent = true
+        } else {
+            // See if all the oneof fields were in order within the (even if there were number gaps).
+            //    message Good {
+            //      oneof o {
+            //        int32 a = 1;
+            //        int32 z = 26;
+            //      }
+            //    }
+            //    message Bad {
+            //      oneof o {
+            //        int32 a = 1;
+            //        int32 z = 26;
+            //      }
+            //      int32 m = 13;
+            //    }
+            let firstIndex = parentFieldNumbersSorted.index(of: first)!
+            var isContinuousInParent = true
+            for i in 0..<fields.count {
+                if fieldsSortedByNumber[i].number != parentFieldNumbersSorted[firstIndex + i] {
```

Comment:
I find this logic a little confusing.  It would be clearer (to me) to use the same logic as you use for extension checks below, namely:  Are any of the parent field numbers between our first and last field numbers?